lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -1260,14 +1260,19 @@ func (d *zfs) HasVolume(vol Volume) bool {
 	return d.checkDataset(d.dataset(vol, false))
 }
 
-// ValidateVolume validates the supplied volume config.
-func (d *zfs) ValidateVolume(vol Volume, removeUnknownKeys bool) error {
-	rules := map[string]func(value string) error{
-		"zfs.blocksize":        validate.Optional(ValidateZfsVolBlocksize(vol)),
+// commonVolumeRules returns validation rules which are common for pool and volume.
+func (d *zfs) commonVolumeRules() map[string]func(value string) error {
+	return map[string]func(value string) error{
 		"zfs.remove_snapshots": validate.Optional(validate.IsBool),
 		"zfs.use_refquota":     validate.Optional(validate.IsBool),
 		"zfs.reserve_space":    validate.Optional(validate.IsBool),
 	}
+}
+
+// ValidateVolume validates the supplied volume config.
+func (d *zfs) ValidateVolume(vol Volume, removeUnknownKeys bool) error {
+	rules := d.commonVolumeRules()
+	rules["zfs.blocksize"] = validate.Optional(ValidateZfsVolBlocksize(vol))
```

Comment:
This should be handled by commonVolumeRules

Reply:
Done.